File: src/main/java/com/gurbanivision/mboparai/codesamples/javascript/SelectionStatements.js
/**
 * node SelectionStatements.js
 * @author boparaim
 */

'use srict';
console.log('Init (using strict mode)');

function SelectionStatements(costShare, search) {
    this.administrativeCostPercentage = costShare;
    this.searchTerm = search;
}

SelectionStatements.prototype.donate = function(amount) {
    console.log('donate $'+amount+' to charity A');
    return amount;
};

SelectionStatements.prototype.showPictures = function(relatedTo) {
    console.log('images of '+relatedTo);
};

SelectionStatements.prototype.oneWayIfStatement = function() {
    if (this.administrativeCostPercentage < 5) {
        this.donate(50);
    }

    if (this.administrativeCostPercentage < 5)
        this.donate(50);
};

SelectionStatements.prototype.twoWayIfStatement = function() {
    if (this.administrativeCostPercentage < 5) {
        this.donate(50);
    }
    else {
        this.donate(30);
    }

    if (this.administrativeCostPercentage < 5)
        this.donate(50);
    else
        this.donate(30);
};

SelectionStatements.prototype.multiWayIfStatement = function() {
    if (this.administrativeCostPercentage < 5) {
        this.donate(50);
    }
    else if (this.administrativeCostPercentage < 10) {
        this.donate(40);
    }
    else {
        this.donate(30);
    }

    if (this.administrativeCostPercentage < 5)
        this.donate(50);
    else if (this.administrativeCostPercentage < 10)
        this.donate(40);
    else
        this.donate(30);
};

SelectionStatements.prototype.switchStatement = function() {
    switch (this.searchTerm) {
        case "dog":
            this.showPictures("dog");
            break;
        case "cat":
            this.showPictures("cat");
            break;
        default:
            this.showPictures("scorpion");
            break;
    }

    switch (this.searchTerm) {
        case "dog":
        case "puppy":
            this.showPictures("dog");
            break;
        default:
            this.showPictures("scorpion");
    }
};

SelectionStatements.prototype.conditionalExpression = function() {
    var donatedAmount = (this.administrativeCostPercentage < 5) ? this.donate(50) : this.donate(40);
        
    donatedAmount = (this.administrativeCostPercentage < 5) ? this.donate(50) 
                                : (this.administrativeCostPercentage < 10) ? this.donate(40) : this.donate(30);
};

var selectionStatements = new SelectionStatements(5, 'dog');
selectionStatements.oneWayIfStatement();
selectionStatements.twoWayIfStatement();
selectionStatements.multiWayIfStatement();
selectionStatements.switchStatement();
selectionStatements.conditionalExpression();

File: src/main/java/com/gurbanivision/mboparai/codesamples/python/SelectionStatements.py
'''
python3 SelectionStatements.py
Created on Feb 22, 2018

@author: boparaim
'''

class SelectionStatements(object):
    '''
    classdocs
    '''

    def __init__(self, costShare, search):
        '''
        Constructor
        '''
        self.administrativeCostPercentage = costShare
        self.searchTerm = search
        
    def donate(self, amount):
        print('donate $'+str(amount)+' to charity A')
        return amount
    
    def showPictures(self, relatedTo):
        print('images of '+relatedTo)
        
    def oneWayIfStatement(self):
        if (self.administrativeCostPercentage < 5):
            self.donate(50);
        
    def twoWayIfStatement(self):
        if (self.administrativeCostPercentage < 5):
            self.donate(50);
        else:
            self.donate(30)
        
    def multiWayIfStatement(self):
        if (self.administrativeCostPercentage < 5):
            self.donate(50);
        elif (self.administrativeCostPercentage < 10):
            self.donate(40);
        else:
            self.donate(30)
    
    # switch case is not available
    # dictionaries can be used instead
    def swithStatement(self):
        output = {
            'dog': self.showPictures,
            'cat': self.showPictures
        }
        try:
            output[self.searchTerm](self.searchTerm)
        except KeyError:
            self.showPictures('scorpion')
    
    def conditionalExpression(self):
        donatedAmount = self.donate(50) if self.administrativeCostPercentage < 5 else self.donate(40)
        donatedAmount = self.donate(50) if self.administrativeCostPercentage < 5 else self.donate(40) if self.administrativeCostPercentage < 10 else self.donate(30)
        
    
selectionStatements = SelectionStatements(5, 'dog')
selectionStatements.oneWayIfStatement()
selectionStatements.twoWayIfStatement()
selectionStatements.multiWayIfStatement()
selectionStatements.swithStatement()
selectionStatements.conditionalExpression()

    
    
        